aws/src/test/java/org/apache/iceberg/aws/TestAwsClientFactories.java

```diff
@@ -132,6 +136,41 @@ public void testLakeFormationAwsClientFactorySerializable() throws IOException {
         .isInstanceOf(LakeFormationAwsClientFactory.class);
   }
 
+  @Test
+  public void testDefaultAwsClientFactoryWithCredentialsProvider() {
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put(AwsProperties.AWS_REGION, Region.AWS_GLOBAL.toString());
+    properties.put(
+        AwsProperties.S3FILEIO_CREDENTIALS_PROVIDER,
+        SystemPropertyCredentialsProvider.class.getName());
+
+    AwsClientFactory defaultAwsClientFactory = AwsClientFactories.from(properties);
+    Assertions.assertThat(defaultAwsClientFactory)
+        .isInstanceOf(AwsClientFactories.DefaultAwsClientFactory.class);
+
+    try (S3Client s3Client = defaultAwsClientFactory.s3()) {
+      Assertions.assertThat(s3Client != null);
+      URL url =
+          s3Client
+              .utilities()
+              .getUrl(GetUrlRequest.builder().bucket("test-bucket").key("test-key").build());
+      Assert.assertTrue(
+          url != null && url.toString().equals("https://test-bucket.s3.amazonaws.com/test-key"));
+    }
+  }
+
+  @Test
+  public void testDefaultAwsClientFactoryWithInvalidCredentialsProvider() {
+    Map<String, String> properties = Maps.newHashMap();
+    properties.put(AwsProperties.AWS_REGION, Region.AWS_GLOBAL.toString());
+    properties.put(AwsProperties.S3FILEIO_CREDENTIALS_PROVIDER, "invalidClassName");
+    AssertHelpers.assertThrows(
+        "Should fail if invalid credentials provider set",
+        IllegalArgumentException.class,
+        "The creation of an instance of invalidClassName failed",
+        () -> AwsClientFactories.from(properties).s3().close());
```

Comment:
[suggestion] Since there is another PR (#6977) planning on deprecating `AssertHelpers`, shall we use `Assertions.assertThatThrownBy` here instead?

It may be something like:
```java
Assertions.assertThatThrownBy(() -> AwsClientFactories.from(properties).s3().close())
.isInstanceOf(IllegalArgumentException.class)
.hasMessageContaining("The creation of an instance of invalidClassName failed");
```

Reply:
Thanks. I replaced it with your suggestion. Please check